crates/uv-installer/src/compile.rs

```diff
@@ -85,12 +86,34 @@ pub async fn compile_tree(
     debug!("Starting {} bytecode compilation workers", worker_count);
     let mut worker_handles = Vec::new();
     for _ in 0..worker_count.get() {
-        worker_handles.push(tokio::task::spawn(worker(
+        let (tx, rx) = oneshot::channel();
+
+        let worker = worker(
             dir.to_path_buf(),
             python_executable.to_path_buf(),
             pip_compileall_py.clone(),
             receiver.clone(),
-        )));
+        );
+
+        // Spawn each worker on a dedicated thread.
+        std::thread::Builder::new()
+            .name("uv-compile".to_owned())
+            .spawn(move || {
+                // Report panics back to the main thread.
+                let result = panic::catch_unwind(AssertUnwindSafe(|| {
```

Comment:
I'm probably missing something silly here, but why the `catch_unwind`? You have a thread boundary here, so you should be able to just extract the panic from the result of joining the thread?

Reply:
We wait on the threads asynchronously, which is why we use the oneshot channel instead of blocking on `join`.

Reply:
Hmmm okay, I think that makes sense to me. Thanks!